messages/jsonschema/scripts/templates/php.php.erb

```diff
@@ -0,0 +1,88 @@
+<?php declare(strict_types=1);
+
+/**
+ * This code was auto-generated by {this script}[https://github.com/cucumber/common/blob/main/messages/jsonschema/scripts/codegen.rb]
+ */
+
+namespace Cucumber\Messages;
+
+use \JsonSerializable;
+use Cucumber\Messages\DecodingException\SchemaViolationException;
+
+<%- @schemas.sort.each do |key, schema| -%>
+// CLASS_START <%= class_name(key) %>.php
+/**
+ * Represents the <%= class_name(key) %> message in Cucumber's message protocol
+ * @see https://github.com/cucumber/common/tree/main/messages#readme
+ *
+<%=- format_description(schema['description'], indent_string: '') -%> */
+final class <%= class_name(key) %> implements JsonSerializable
+{
+    use JsonEncodingTrait;
+
+    private function __construct(
```

Comment:
In retrospect this should be public, otherwise gherkin implementation won't be able to do much useful

Reply:
Indeed, in other implementation, we instantiate messages directly without de-serializing those from arrays or json.
It would be more efficient to do the same with php I suppose.

Reply:
They're now able to be constructed (see README)